Stack/IStack.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+template <typename ItemType>
+class IStack
+{
+public:
+	virtual ~IStack() { }
+
+	/// <summary>
+	/// Add an item to the stack, as the new top
+	/// </summary>
+	/// <param name="item"></param>
+	virtual void Push(const ItemType& item) = 0;
+
+	/// <summary>
+	/// Takes an item from the stack's top
+	/// </summary>
+	/// <param name="defaultValue">Return value in case the stack is empty</param>
+	/// <returns>If not empty, returns the poped item, else, returns defaultValue</returns>
+	virtual ItemType Pop(const ItemType& defaultValue = ItemType()) = 0;
+
+	/// <summary>
+	/// Just peek's the top item. If empty, returns defaultValue
+	/// </summary>
+	/// <param name="defaultValue">Value to return if stack is empty</param>
+	/// <returns>The top item</returns>
+	virtual ItemType Top(const ItemType& defaultValue = ItemType()) const = 0;
```

Comment:
I have 2 questions here: 
1) Why Top and Pop have parametheres? They don't need it.
2) What const ItemType& defaultValue = ItemType() does? I have never seen the expresion after the =. It's like calling a constructor of the type? I dont understand it.